src/services/completions.ts

```diff
@@ -3754,6 +3754,12 @@ namespace ts.Completions {
             // Object literal is assignment pattern: ({ | } = x)
             return typeChecker.getTypeAtLocation(node.parent);
         }
+        if (isParenthesizedExpression(node.parent)) {
+            const parent = walkUpParenthesizedExpressions(node.parent);
+            if (isExpression(parent)) {

```

Comment:
I think it would be better to walk up parentheses at the beginning of the function, then run *both* of the existing checks.
At worst it's written the way the rest of the compiler is, at best it allows parentheses inside assignment pattern + future things that might be added here.

Reply:
I've moved `walkUpParenthesizedExpressions`. Using parentheses inside assignment patterns is forbidden, so this is still more related to the second case where we need to get `contextualType` of expression.